modules/core/src/test/java/org/apache/ignite/internal/metric/SystemViewSelfTest.java

```diff
@@ -461,6 +464,63 @@ public void testClientsConnections() throws Exception {
         }
     }
 
+    /** */
+    @Test
+    public void testNodes() throws Exception {
+        try(IgniteEx g1 = startGrid(0)) {
+            SystemView<ClusterNodeView> nodes = g1.context().systemView().view(NODES_SYS_VIEW);
+
+            assertEquals(1, nodes.size());
+
+            checkNodeView(nodes.iterator().next(), g1.localNode(), true);
+
+            try(IgniteEx g2 = startGrid(1)) {
+                awaitPartitionMapExchange();
+
+                assertEquals(2, nodes.size());
+
+                boolean found = false;
+
+                for (ClusterNodeView node : nodes) {
+                    if (!node.nodeId().equals(g2.localNode().id()))
+                        continue;
+
+                    checkNodeView(node, g2.localNode(), false);
+
+                    found = true;
+
+                    break;
+                }
+
+                assertTrue(found);
+
+                SystemView<ClusterNodeView> nodes2 = g2.context().systemView().view(NODES_SYS_VIEW);
+
+                assertEquals(2, nodes2.size());
+
+                for (ClusterNodeView node : nodes2) {
+                    if(node.nodeId().equals(g2.localNode().id()))
+                        checkNodeView(node, g2.localNode(), true);
+                    else
+                        checkNodeView(node, g1.localNode(), false);
+                }
+            }
+        }
```

Comment:
we should check that views size is 0

Reply:
View size can't be 0, because we have alive node.
`assertEquals(1, views.size());` added.